applications/asset_tracker_v2/src/events/location_module_event.h

```diff
@@ -0,0 +1,164 @@
+	 *  the event struct member ``data.location``.
+	 *  The ``data.location`` struct members ``search_time`` and ``satellites_tracked``
+	 *  contain valid data.
+	 */
+	LOCATION_MODULE_EVT_TIMEOUT,
+
+	/** An active location search has started. */
+	LOCATION_MODULE_EVT_ACTIVE,
+
+	/** A location search has stopped, either as a result of timeout, error
+	 *  or acquired location.
+	 */
+	LOCATION_MODULE_EVT_INACTIVE,
+
+	/** The module has been shut down gracefully.
+	 *  The event has associated payload of the type ``uint32`` in the struct member
+	 *  ``data.id`` that contains the module ID used when acknowledging a shutdown
+	 *  request from the util module.
+	 */
+	LOCATION_MODULE_EVT_SHUTDOWN_READY,
+
+	/** The location library has reported that it needs GPS assistance data.
+	 *  The event has associated payload of the type ``struct nrf_modem_gnss_agps_data_frame``
+	 *  in the event struct member ``data.agps_request``, which contains the types
+	 *  of A-GPS data that the modem needs.
+	 */
+	LOCATION_MODULE_EVT_AGPS_NEEDED,
+
+	/** The location library has reported that it needs GPS prediction data.
+	 *  The event has associated payload of the type ``struct gps_pgps_request``
+	 *  in the event struct member ``data.pgps_request``, which specifies the P-GPS data need.
+	 */
+	LOCATION_MODULE_EVT_PGPS_NEEDED,
+
+	/** An error has occurred, and data may have been lost.
+	 *  The event has associated payload of the type ``int`` in the struct member
+	 *  ``data.err``, which contains the original error code that triggered
+	 *  the sending of this event.
+	 *
+	 */
+	LOCATION_MODULE_EVT_ERROR_CODE,
+};
+
+/** @brief Position, velocity and time (PVT) data. */
+struct location_module_pvt {
+	/** Longitude in degrees. */
+	double longitude;
+
+	/** Latitude in degrees. */
+	double latitude;
+
+	/** Altitude above WGS-84 ellipsoid in meters. */
+	float altitude;
+
+	/** Position accuracy (2D 1-sigma) in meters. */
+	float accuracy;
+
+	/** Horizontal speed in m/s. */
+	float speed;
+
+	/** Heading of user movement in degrees. */
+	float heading;
+};
+
+struct location_module_neighbor_cells {
+	struct lte_lc_cells_info cell_data;
+	struct lte_lc_ncell neighbor_cells[17];
+	int64_t timestamp;
+};
+
+/** @brief Location module data for associated payload for events of LOCATION_MODULE_EVT_TIMEOUT and
+ *	   LOCATION_MODULE_EVT_DATA_READY types.
+ */
+struct location_module_data {
+	/** Location data in the form of a PVT structure. */
+	struct location_module_pvt pvt;
+
+	uint8_t satellites_tracked;
+
+	/** Time from the search was initiated until fix or timeout occurred. */
+	uint32_t search_time;
+	int64_t timestamp;
+};
+
+/** @brief Location module event. */
+struct location_module_event {
+	struct app_event_header header;
+	enum location_module_event_type type;
+
+	union {
+		struct location_module_data location;
+		struct location_module_neighbor_cells neighbor_cells;
+		struct nrf_modem_gnss_agps_data_frame agps_request;
```

Comment:
will you protect this with #if defined(CONFIG_NRF_CLOUD_PGPS) if A-GPS can be disabled?

Reply:
You mean `#if defined(CONFIG_NRF_CLOUD_AGPS)`?
That's added now.

Reply:
I need to check this later. Don't close this comment.

Reply:
We can revisit this in NCSDK-18093 if needed although discussion in https://github.com/nrfconnect/sdk-nrf/pull/8631#discussion_r1013943036 is going towards not adding flagging in the header file.

Reply:
There is now flagging for PGPS but not for AGPS. The reason for this is that PGPS is not included into the unit tests. Otherwise I could remove that flagging also.